### PR TITLE
JS: fix yaml StringIndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by esben-semmle**
This fixes ODASA-7993.

We previously got a `StringIndexOutOfBoundsException` in `mkToString` on some YAML files with long lines that had been truncated before we got access to them.

We no longer get an exception on such truncated lines, and treat them like multiline strings.

I think this needs to go into 1.21.

@tibbes

### CODE REVIEW

File: javascript/extractor/src/com/semmle/js/extractor/YAMLExtractor.java [hunk 1]
```diff
@@ -226,16 +226,19 @@ private String getTag(String explicitTag, NodeId kind, String value, boolean imp
...
    */
   private String mkToString(Mark startMark, Mark endMark) {
     String snippet = startMark.get_snippet(0, Integer.MAX_VALUE);
     int nl = snippet.indexOf('\n');
     String context = snippet.substring(0, nl);
     String src = context.substring(snippet.substring(nl + 1).indexOf('^'));
-    if (endMark.getLine() == startMark.getLine())
-      src = src.substring(0, endMark.getColumn() - startMark.getColumn());
+    int desiredSingleLineSrcLength = endMark.getColumn() - startMark.getColumn();
+    boolean isShortString = src.length() >= desiredSingleLineSrcLength;
```
Comment (asger-semmle) [inline javascript/extractor/src/com/semmle/js/extractor/YAMLExtractor.java]: I'm confused by the name `isShortString` when it's defined to be true is that `src` is *longer* than a certain limit. Was there a negation that was flipped during a refactoring here?

Comment (esben-semmle) [reply javascript/extractor/src/com/semmle/js/extractor/YAMLExtractor.java]: Amended to `hasAccessToDesiredString`.

**Comment by xiemaisi** [comment]
> I think this needs to go into 1.21.

It's targeted at master, though.

**Comment by xiemaisi** [comment]
If we're putting this into 1.21, then maybe we should include https://github.com/Semmle/ql/pull/1481 as well. What do you think?

**Comment by esben-semmle** [comment]
I agree, as they are both simple crash fixes. Do you want a cherry pick of the #1481 commit into this PR, or a separate PR?

**Comment by xiemaisi** [comment]
Maybe just add it to this PR?

**Comment by xiemaisi** [comment]
This PR may not need to go into the release after all, so please don't merge yet.

**Pull request merged**